client/state/data-layer/README.md

```diff
@@ -0,0 +1,116 @@
+They should _trust_ the framework to make sure those needs are met if indeed it is possible to meet them.
+
+In contrast to the prior solutions the Redux actions which components will end up dispatching do not directly invoke the associated API calls and asynchronous fetches.
+Middleware at this layer will trap those actions and interpret them however is most fitting.
+For example, it may determine that an update to one type of data justifies also updating another related type of data at the same time.
+It may determine that some actions, such as submitting a post or comment, justify waiting a short period of time before actually requesting an update on the server to allow for cancelling accidental submissions.
+
+This middleware should break the tight coupling between Calypso's internal data structure and that presented by the APIs or backends which supply it.
+Components should ideally not need to know that some parts of related data (such as a site's settings or plan subscriptions) comes from distinct API endpoints.
+Instead the components should be able to request that they require such data and the middleware will intercept the request to fulfill it accordingly.
+
+## Implementation
+
+The data layer _intercepts_ Redux actions.
+Once it has performed its behavior, this action will be dropped entirely from the dispatch path.
+This is to prevent two middlewares which can both supply a given request from fighting with each other or duplicating the request; consequently it allows for prioritization of data-fetching needs by means of ordering how the middleware are arranged in the chain. For example, we could start polling from the WP-API for posts but leave all other requests up to the WordPress.com API.
+Note that we can still allow multiple functions within the same middleware to handle the same action.
+The WordPress.com API middleware demonstrates this in building a tree of handlers which can all respond to given action types.
+
+Each middleware intercepts given Redux actions and will correspondingly dispatch new follow-up actions to actually handle their requests.
+The functions that compose to form the middleware _can_ and in most cases _will_ closely resemble what was previously written in `redux-thunk` actions.
+
+The middleware should be pieced together from multiple handler functions in a logical way which encourages modularity and isolation of responsibilities.
+The data layer provides `mergeHandlers()` as a utility to support this.
+**Each file should export an object whose keys are Redux action types to intercept and whose values are lists of functions to invoke on those actions.**
+The list of handlers will be called in sequence in the order given in the list itself and in the order in which the different handlers are merged.
+Even when there is only one handler the value of the exported key should be a list - a singleton list in this case.
+
+```js
```

Comment:
Should we add comment explaining that this part lists content from `splineEndpoint.js`? This is at least how I understood it :)